06.event_listeners/src/eventListeners.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-console */
+interface EventsMap {
+  events: Map<string, CallbackData>
+}
+
+interface CallbackData {
+  callbacks: EventListener[],
+  eventHandler: EventListener,
+}
+
+const elementMap: Map<HTMLElement, EventsMap> = new Map<HTMLElement, EventsMap>();
+
+function eventHandler(event: Event) {
+  // eslint-disable-next-line prefer-destructuring
+  const target = <HTMLElement>event.target;
+  if (target && elementMap.has(target) && elementMap.get(target)?.events.has(event.type)) {
+    const callbackArray = elementMap.get(target)!.events.get(event.type);
+    callbackArray!.callbacks.forEach((callback) => {
+      callback(event);
+    });
+  } else {
+    console.warn(`EventHandler doesn't have the required data for event: "${event.type}" on element: ${target.id}`);
+  }
+}
+
+function removeEvent(
+  element: HTMLElement,
+  event: string,
+  eventHandlerEvn: EventListener,
+) {
+  if (element && event && eventHandlerEvn) {
+    element.removeEventListener(event, eventHandlerEvn);
+  } else {
+    console.warn(`RemoveEvent is missing required data, element: ${element.id}, event: "${event}" or eventHandler`);
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const eventListener = {
+  on(element: HTMLElement, event: string, callback: EventListener) {
+    if (!elementMap.has(element)) {
+      elementMap.set(element, {
+        events: new Map([[event, { callbacks: [callback], eventHandler }]]),
+      });
+    } else {
+      const mapEvent = elementMap.get(element)!.events;
+      if (!mapEvent.has(event)) {
+        mapEvent.set(event, { callbacks: [callback], eventHandler });
+      } else {
+        mapEvent.get(event)!.callbacks.push(callback);
+      }
+    }
+    element.addEventListener(event, eventHandler);
+  },
+
+  off(element: HTMLElement, event?: string, callback?: EventListener) {
+    if (elementMap.has(element)) {
+      const eventMap = elementMap.get(element);
+      if (event) {
+        const callbackData = eventMap!.events.get(event);
+        if (callbackData) {
+          if (callback) {
+            const index = callbackData.callbacks.indexOf(callback, 0);
+            if (index > -1) {
+              callbackData.callbacks.splice(index, 1);
+              if (callbackData.callbacks.length === 0) {
+                eventMap!.events.delete(event);
+                removeEvent(element, event, callbackData.eventHandler);
```

Comment:
here you can just removeEvent(element, event) - to remove everything for that event.